README.md

```diff
@@ -1,119 +1,231 @@
+preventing any subsequent filter from being executed on the value.
+Otherwise, it will pass the value as it is to the next filter.
+
+
+### filters.required
+Throws if the value is `null` or `undefined`.
+
+
+### filters.array(filters...)
+```javascript
+var input = {
+  aListOfStuff: [1, 4, 5]
+}
+
+var schema = {
+  list: ['aListOfStuff', filters.array(filters.isInteger, function(n){return n+'.0'})]
+}
+
+var expectedOutput = {
+  list: ['1.0', '4.0', '5.0']
+}
+
+assert.deepEqual(cartography.map(input, schema), expectedOutput)
 ```
-inputUser =
-  id: 3
-  email: 'joe.lizard@reptiles.com'
-  password: '12345'
-  sales:
-    saleCount: 56
-    salePercentage: 23
-  addressStreet: 'rock st'
-  addressNumber: '21/a'
-  addressState: 'VIC'
-
-outputUser = map cartographyUserSchema, inputUser
-
- * outputUser will yield:
-  id: 3
-  username: 'joe.lizard@reptiles.com'
-  password: '12345'
-  provider: 'Adslot'
-  maxShare: 23
-  address:
-    streetName: 'rock st'
-    streetNumber: '21/a'
-    state: 'VIC'
+`filters.array` ensures that the value is an Array and applies the specified `filters`, if any, to each element of the value.
+If `filters` contain any nested Array, they will be flattened.
+
+
+### filters.object(schema)
+```javascript
+var input = {
+  someHash: {
+    a: 1,
+    b: 'hello!',
+    c: ['blue', 'black'],
+  }
+}
+
+var schema = {
+  keys: ['someHash', filters.object({
+    a: [filters.isNumber],
+    b: [filters.isString, function(s) {return '**'+ s +'**'}],
+    c: [filters.array(filters.isString)]
+  })]
+}
+
+var expectedOutput = {
+  keys: {
+    a: 1,
+    b: '**hello!**',
+    c: ['blue', 'black']
+  }
+}
+
+assert.deepEqual(map(input, schema), expectedOutput)
 ```
+`filters.object` creates a filter that passes the value through `cartography.map` with the given nested schema.
 
-If `inputUser` contained no `sales` attribute or no `salePercentage` within
-`sales`, `map()` would throw a `CartographyError` with a
-"sales.salePercentage: is required" `message`.
+
+### filters.assert(condition, errorMessage)
+```javascript
+var isEven = filters.assert(function(n){return n % 2 === 0}, 'must be even')
+```
+Returns a filter that asserts for the given condition, producing a `CartographyError` with the provided
+message if the condition is not met.
 
 
-Asynchronous version
---------------------
-`asyncMap` is a fully asynchronous version of `map`, which will return
-`CartographyError`s to the provided callback instead of throwing them.
-In addition to the usual synchronous filters and custom functions, `asyncMap`
-can use asynchronous filters and asynchronous custom functions provided they
-are wrapped by the `async` function decorator.
-The `asyncArray` filter should also be used in place of `array`.
+### filters.parseJSON
+Converts a JSON string into a JavaScript Object.
 
 
+### filters.isString
+Throws if the value is not a string (will reject String objects).
```

Comment:
This isn't clear to me. If it rejects `String` objects does it only accept string literals? Might be good to call that out explicitly for people that aren't as well versed in the joys of JavaScript